src/Language/PureScript/Sugar/Names/Common.hs

```diff
@@ -24,14 +24,45 @@ warnDuplicateRefs
   -> m ()
 warnDuplicateRefs pos toError refs = do
   let withoutCtors = deleteCtors `map` refs
-      dupeRefs = mapMaybe (refToName pos) $ withoutCtors \\ nub withoutCtors
+      dupeRefs = mapMaybe (refToName pos) $ onlyDuplicates withoutCtors
       dupeCtors = concat $ mapMaybe (extractCtors pos) refs
 
   for_ (dupeRefs ++ dupeCtors) $ \(pos', name) ->
     warnWithPosition pos' . tell . errorMessage $ toError name
 
   where
 
+  -- Finds all duplicate refs
+  --onlyDuplicates :: forall a. Eq a => Ord a => [a] -> [a]
+  onlyDuplicates :: Eq a => Ord a => [a] -> [a]
```

Comment:
If I'm reading this right... could this be `concatMap (drop 1) . group . sort` ?

Reply:
Nice, that's certainly a lot more elegant! Your version matches the original `\\ nub` behavior, which discards one of the duplicated elements. I think it's better to preserve all duplicates. Here's a version that does that:
```hs
onlyDuplicates :: Eq a => Ord a => [a] -> [a]
onlyDuplicates = concat . filter twoOrMore . group . sort
  where
    twoOrMore :: [a] -> Bool
    twoOrMore [] = False
    twoOrMore [x] = False
    twoOrMore (x : _) = True
```
Figure that a custom `twoOrMore` function is faster than `(\x -> length x > 1)` with `length` being `O(n)`.

Reply:
> Figure that a custom twoOrMore function is faster than (\x -> length x > 1) with length being O(n).

There's no big `n`'s unless you're actually duplicating the same name _a lot_ so I wouldn't bother. You can check your benchmark but I doubt it's going to be noticeable.